openedx/features/course_experience/utils.py

```diff
@@ -156,13 +156,13 @@ def mark_last_accessed(user, course_key, block):
     course_outline_root_block = all_blocks['blocks'].get(all_blocks['root'], None)
     if course_outline_root_block:
         populate_children(course_outline_root_block, all_blocks['blocks'])
-        set_last_accessed_default(course_outline_root_block)
-
-        mark_blocks_completed(
-            block=course_outline_root_block,
-            user=request.user,
-            course_key=course_key
-        )
+        if user:
```

Comment:
An anonymous user object should not be passed here since that does not have an actual user. I am going to add an assert to the top of this function instead.

Reply:
👍